### PR TITLE
Move graph deletion to datapipe

### PR DESCRIPTION
**Pull request by rvazarkar**
## Description
Moves logic for graph deletion to the datapipe to prevent deadlocks

## Motivation and Context
The existing logic for clearing the DB was handled by the api handler in resources. However, if concurrent graph modifications were being performed in datapipe, this would lead to graph deadlocks. This MR moves the graph deletion logic into datapipe

https://specterops.atlassian.net/browse/BED-4286
## How Has This Been Tested?

Local testing

## Types of changes

-   [ ] Chore (a change that does not modify the application functionality)
-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [ ] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: cmd/api/src/daemons/datapipe/delete.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+			// deleting a node also deletes all of its edges due to a sql trigger
+			if err := batch.DeleteNode(nodeId); err != nil {
+				return err
+			}
+		}
+		return nil
+	}); err != nil {
+		return fmt.Errorf("error deleting all nodes: %w", err)
+	} else {
+		// if successful, handle audit log and kick off analysis
```
Comment (juggernot325) [inline cmd/api/src/daemons/datapipe/delete.go:33]: This comment is a little confusing. I would expect the code to perform these operations to live here based on the wording. Am I missing something?

File: justfile [hunk 2]
```diff
@@ -99,7 +99,7 @@ update-favicon:
 
 # run go commands in the context of the api project
 go *ARGS:
-  @cd cmd/api/src && GODEBUG=cgocheck=2 go {{ARGS}}
+  @cd cmd/api/src && go {{ARGS}}
```
Comment (juggernot325) [inline justfile:102]: Was this change intentional? It seems out of scope from the original issue.

File: cmd/api/src/daemons/datapipe/delete.go (hunk 1 above)
Comment (rvazarkar) [reply cmd/api/src/daemons/datapipe/delete.go:33]: leftover comment from copypaste

File: justfile (hunk 2 above)
Comment (rvazarkar) [reply justfile:102]: It is out of scope, but that flag is deprecated and keeps just go * commands from running so I just fixed it while testing

**Review by juggernot325 — APPROVED** [review]
I did have a couple questions, but it's likely they won't require code changes. I'm preemptively approving under the assumption that nothing further will be needed. If that changes I'll revoke the approval.

File: cmd/api/src/daemons/datapipe/datapipe.go [hunk 3]
```diff
@@ -182,6 +206,25 @@ func (s *Daemon) Start(ctx context.Context) {
...
 }
 
+func (s *Daemon) deleteData() {
+	defer func() {
+		s.status.Update(model.DatapipeStatusIdle, false)
+		s.setDeletionRequested(false)
+		s.setAnalysisRequested(true)
+	}()
+	defer log.Measure(log.LevelInfo, "Purge Graph Data Completed")()
+	s.status.Update(model.DatapipeStatusPurging, false)
```
Comment (brandonshearin) [inline cmd/api/src/daemons/datapipe/datapipe.go:216]: should this be setting the purging state to true?

Comment (superlinkx) [reply cmd/api/src/daemons/datapipe/datapipe.go:216]: This method has a confusing signature if you're not looking at it in an IDE. The bool flag is to say whether to update analysis time, which we don't want to do because this isn't an analysis operation. Regardless of what you set that flag to, the status you give it will become the current status for the datapipe.

**Review by brandonshearin — APPROVED** [review]
